src/components/contractualAgreement/ContractualAgreement.tsx

```diff
@@ -15,8 +16,19 @@ interface Props {
 }
 
 export default function ContractualAgreement({ offerId, offerData }: Props) {
-  // TODO: get the template from the offer metadata (BP390 - https://app.asana.com/0/1200803815983047/1203080300620356)
-  const templateUrl = CONFIG.buyerSellerAgreementTemplate as string;
+  const { data: offerFetched, isFetching } = useOffer(
```

Comment:
why do we need to fetch the offer data again, if offerData already contains the information we need?

Reply:
sometimes offerData is passed in undefined, sometimes the offerId is undefined...so depending on where this is called from @levalleux-ludo

Reply:
changed, thanks! @levalleux-ludo